src/table/TableHeadWrapper.jsx

```diff
@@ -65,22 +77,13 @@ export default function TableHeadWrapper({
                 )
               }
               onMouseDown={() => handleClickToFocusHead(columnIndex, focusedCellCoord, rootElement)}
+              onClick={() =>
+                !selectionsAPI.isModal() && !constraints.active && changeSortOrder(layout, column.isDim, columnIndex)
+              }
             >
-              <TableSortLabel
-                active={layout.qHyperCube.qEffectiveInterColumnSortOrder[0] === columnIndex}
-                direction={column.sortDirection}
-                onClick={() =>
-                  // when cells are selected or in edit mode, it should not be able to do the sorting
-                  !selectionsAPI.isModal() && !constraints.active && changeSortOrder(layout, column.isDim, columnIndex)
-                }
-                tabIndex={-1}
-              >
+              <TableSortLabel active={isCurrentColumnActive} direction={column.sortDirection} tabIndex={-1}>
                 {column.label}
-                {layout.qHyperCube.qEffectiveInterColumnSortOrder[0] === columnIndex ? (
-                  <span className={classes.visuallyHidden}>
-                    {column.sortDirection === 'desc' ? 'sorted descending' : 'sorted ascending'}
-                  </span>
-                ) : null}
+                <span className={classes.visuallyHidden}>{currentSortDir}. Press space to sort on this column.</span>
```

Comment:
No need to have space in the string, it is implied since it is a button. Also, need to take `isCurrentColumnActive` into account. Shouldn't announce ascending/descending if `isCurrentColumnActive` is false

Reply:
no I meant the word `space`. A button should listen to both space and enter. And since the screen reader says button, no need to explain what key to use